src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1148,21 +1147,33 @@ internal SingleLookupResult CheckViability(Symbol symbol, int arity, LookupOptio
                 return LookupResult.Inaccessible(symbol, diagInfo);
             }
             else if (!InCref &&
-                     !this.IsAccessible(unwrappedSymbol,
+                    !this.IsAccessible(unwrappedSymbol,
                                         RefineAccessThroughType(options, accessThroughType),
                                         out inaccessibleViaQualifier,
                                         ref useSiteDiagnostics,
                                         basesBeingResolved))
             {
-                if (inaccessibleViaQualifier)
-                {
-                    diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadProtectedAccess, unwrappedSymbol, accessThroughType, this.ContainingType) : null;
-                }
-                else
-                {
-                    var unwrappedSymbols = ImmutableArray.Create<Symbol>(unwrappedSymbol);
-                    diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadAccess, new[] { unwrappedSymbol }, unwrappedSymbols, additionalLocations: ImmutableArray<Location>.Empty) : null;
-                }
+                 bool friendRefNotEqualToThis = unwrappedSymbol != null && unwrappedSymbol.ContainingAssembly.GetInternalsVisibleToPublicKeys(this.ContainingType.ContainingAssembly.Name).Any()
```

Comment:
I don't think `unwrappedSymbol` can be null here. It looks like it's always assigned above.